tls/client-tls-pkcallback.c

```diff
@@ -0,0 +1,367 @@
+}
+
+static int myEccSign(WOLFSSL* ssl, const byte* in, word32 inSz,
+        byte* out, word32* outSz, const byte* key, word32 keySz, void* ctx)
+{
+    int ret;
+    byte* keyBuf = (byte*)key;
+    PkCbInfo* cbInfo = (PkCbInfo*)ctx;
+
+    printf("PK ECC Sign: inSz %u, keySz %u\n", inSz, keySz);
+
+#ifdef WOLFSSL_ASYNC_CRYPT
+    if (cbInfo->state == 0) {
+        cbInfo->state++;
+        printf("PK ECC Sign: Async Simulate\n");
+        return WC_PENDING_E;
+    }
+#endif
+
+    ret = load_key_file(cbInfo->keyFile, &keyBuf, &keySz);
+    if (ret == 0) {
+        ret = wc_ecc_init(&cbInfo->key);
+        if (ret == 0) {
+            word32 idx = 0;
+            ret = wc_EccPrivateKeyDecode(keyBuf, &idx, &cbInfo->key, keySz);
+            if (ret == 0) {
+                WC_RNG *rng = wolfSSL_GetRNG(ssl);
+
+                printf("PK ECC Sign: Curve ID %d\n", cbInfo->key.dp->id);
+                ret = wc_ecc_sign_hash(in, inSz, out, outSz, rng, &cbInfo->key);
+            }
+            wc_ecc_free(&cbInfo->key);
+        }
+    }
+    free(keyBuf);
+
+#ifdef WOLFSSL_ASYNC_CRYPT
+    cbInfo->state = 0;
+#endif
+
+    printf("PK ECC Sign: ret %d outSz %u\n", ret, *outSz);
+
+    return ret;
+}
+#endif /* HAVE_PK_CALLBACKS */
+
+int main(int argc, char** argv)
+{
+    int                ret = 0, err;
+    int                sockfd = SOCKET_INVALID;
+    struct sockaddr_in servAddr;
+    char               buff[256];
+    size_t             len;
+
+    /* declare wolfSSL objects */
+    WOLFSSL_CTX* ctx = NULL;
+    WOLFSSL*     ssl = NULL;
+
+    /* PK callback context */
+    PkCbInfo myCtx;
+    memset(&myCtx, 0, sizeof(myCtx));
+    myCtx.keyFile = KEY_FILE;
+
+    /* Check for proper calling convention */
+    if (argc != 2) {
+        printf("usage: %s <IPv4 address>\n", argv[0]);
+        return 0;
+    }
+
+    /* Create a socket that uses an internet IPv4 address,
+     * Sets the socket to be stream based (TCP),
+     * 0 means choose the default protocol. */
+    if ((sockfd = socket(AF_INET, SOCK_STREAM, 0)) == -1) {
+        fprintf(stderr, "ERROR: failed to create the socket\n");
+        ret = -1;
+        goto exit;
+    }
+
+    /* Initialize the server address struct with zeros */
+    memset(&servAddr, 0, sizeof(servAddr));
+
+    /* Fill in the server address */
+    servAddr.sin_family = AF_INET;             /* using IPv4      */
+    servAddr.sin_port   = htons(DEFAULT_PORT); /* on DEFAULT_PORT */
+
+    /* Get the server IPv4 address from the command line call */
+    if (inet_pton(AF_INET, argv[1], &servAddr.sin_addr) != 1) {
+        fprintf(stderr, "ERROR: invalid address\n");
+        ret = -1;
+        goto exit;
+    }
+
+    /* Connect to the server */
+    if ((ret = connect(sockfd, (struct sockaddr*) &servAddr, sizeof(servAddr)))
+         == -1) {
+        fprintf(stderr, "ERROR: failed to connect\n");
+        goto exit;
+    }
+
+    /*---------------------------------*/
+    /* Start of security */
```

Comment:
This seems like a strange comment.  Perhaps "Start of wolfSSL initialization and configuration"

Reply:
Fixed. Also fixed in several other places.